internal/router/router.go

```diff
@@ -1,146 +1,187 @@
+	router.get("/chain/{chainId}/tables/structure/{hash}", systemCtrl.GetTablesByStructureHash, middlewares.WithLogging, middlewares.OtelHTTP("GetTablesByStructureHash"), middlewares.RESTChainID, rateLim) // nolint
+	router.get("/schema/{table_name}", systemCtrl.GetSchemaByTableName, middlewares.WithLogging, middlewares.OtelHTTP("GetSchemaFromTableName"), rateLim)                                                    // nolint
 
-	router.Get("/query", userController.GetTableQuery, middlewares.WithLogging, middlewares.OtelHTTP("GetTableQuery"), rateLim) // nolint
-	router.Get("/version", infraController.Version, middlewares.WithLogging, middlewares.OtelHTTP("Version"), rateLim)          // nolint
+	router.get("/query", userCtrl.GetTableQuery, middlewares.WithLogging, middlewares.OtelHTTP("GetTableQuery"), rateLim) // nolint
+	router.get("/version", infraCtrl.Version, middlewares.WithLogging, middlewares.OtelHTTP("Version"), rateLim)          // nolint
 
 	// Health endpoint configuration.
-	router.Get("/healthz", healthHandler)
-	router.Get("/health", healthHandler)
-
-	return router
+	router.get("/healthz", controllers.HealthHandler)
+	router.get("/health", controllers.HealthHandler)
 }
 
-func healthHandler(w http.ResponseWriter, _ *http.Request) {
-	w.WriteHeader(http.StatusOK)
+func configureAPIV1Routes(
+	router *Router,
+	rateLim mux.MiddlewareFunc,
+	systemCtrl *controllers.SystemController,
+	userCtrl *controllers.UserController,
+	infraCtrl *controllers.InfraController,
+) error {
+	handlers := map[string]struct {
+		handler     http.HandlerFunc
+		middlewares []mux.MiddlewareFunc
+	}{
+		"QueryFromQuery": {
+			userCtrl.GetTableQuery,
+			[]mux.MiddlewareFunc{middlewares.WithLogging, rateLim},
+		},
+		"ReceiptByTransactionHash": {
+			systemCtrl.GetReceiptByTransactionHash,
+			[]mux.MiddlewareFunc{middlewares.WithLogging, middlewares.RESTChainID, rateLim},
+		},
+		"GetTableById": {
+			systemCtrl.GetTable,
+			[]mux.MiddlewareFunc{middlewares.WithLogging, middlewares.RESTChainID, rateLim},
+		},
+		"Version": {
+			infraCtrl.Version,
+			[]mux.MiddlewareFunc{middlewares.WithLogging, rateLim},
+		},
+		"Health": {
+			controllers.HealthHandler,
+			[]mux.MiddlewareFunc{middlewares.WithLogging, rateLim},
+		},
+	}
+
+	var specRoutesCount int
+	if err := apiv1.NewRouter().Walk(func(route *mux.Route, _ *mux.Router, _ []*mux.Route) error {
+		routeName := route.GetName()
+		// Ignore the "Index" API that OpenAPI 3.0 code generator always create for the base `/` route.
+		if routeName == "Index" {
+			return nil
+		}
+
+		specRoutesCount++
+		endpoint, ok := handlers[routeName]
+		if !ok {
+			return fmt.Errorf("route with name %s not found in handler", routeName)
+		}
+		pathTemplate, err := route.GetPathTemplate()
+		if err != nil {
+			return fmt.Errorf("get path template: %s", err)
+		}
+
+		router.get(
```

Comment:
You are calling `router.get` here because all endpoints defined in [`routes`](https://github.com/tablelandnetwork/go-tableland/pull/412/files#diff-be4fcea6bb23790b7d2da94d48596eec4c96e6c507124ad110278d08b658fc50R50) respond to the `GET` method, right? 

Looks like to me that if we generate a `POST` route, this would code would not work correctly or I'm missing something

Reply:
Correct, since all the methods are GET, this is fine.
Whenever another verb appears, we can use `.GetMethods()` and extend this.

But technically, that should happen in a v2 that would be new code. v1 in theory can't have those kind of changes so this `.get(..)` should be safe to be permanent.